zorg/buildbot/builders/LLDBBuilder.py

```diff
@@ -50,9 +53,18 @@ def getLLDBCMakeBuildFactory(
     lit_args = '-v'
 
     if jobs:
-        build_cmd.append(WithProperties("-j%s" % jobs))
-        install_cmd.append(WithProperties("-j%s" % jobs))
-        test_cmd.append(WithProperties("-j%s" % jobs))
+        jobs_option = ["-j{}".format(jobs)]
+    else:
+        # Use the worker property if option was not explicitly specified.
+        jobs_option = [
+            # If jobs exists, add -j. If not, add nothing.
+            WithProperties("%(jobs:+-j)s"),
```

Comment:
Does `WithProperties` always get the worker properties? Or is there only one set of properties and `jobs` happen to be set by workers?

Reply:
It's the latter. https://github.com/buildbot/buildbot/blob/6e7218792bc324584af980bd1abb4d8b6a5055b8/master/docs/manual/configuration/properties.rst#properties shows all the things that can set properties, and the last one in wins.

```
If the same property is supplied in multiple places, the final appearance takes precedence. For example, a property set in a builder configuration will override the one supplied by the scheduler.
```

So in this case, a build step can override the builder, which can override the worker.

Reply:
Thanks for the explanation. This part is now clear to me.